src/utiltime.cpp

```diff
@@ -32,13 +34,13 @@ void SetMockTime(int64_t nMockTimeIn)
 int64_t GetTimeMillis()
 {
     return std::chrono::duration_cast<std::chrono::milliseconds>(
-            std::chrono::system_clock::now().time_since_epoch()).count();
+            std::chrono::system_clock::now().time_since_epoch() - Epoch).count();
```

Comment:
``time_since_epoch()`` returns a [duration](http://en.cppreference.com/w/cpp/chrono/duration), and ``from_time_t(...)`` [returns](http://en.cppreference.com/w/cpp/chrono/system_clock/from_time_t) a [time_point](http://en.cppreference.com/w/cpp/chrono/time_point). Neither of these are ``int64_t``. I don't see where a coercion to ``int64_t`` is specified in the docs; it might just work by coincidence.

I think it should be something like:
```
static const auto UnixEpoch = std::chrono::system_clock::from_time_t(0);
...

int64_t GetTimeMillis()
{
    return std::chrono::duration_cast<std::chrono::milliseconds>(
        std::chrono::system_clock::now() - UnixEpoch);
}

int64_t GetTimeMicros()
{   
    return std::chrono::duration_cast<std::chrono::microseconds>(
        std::chrono::system_clock::now() - UnixEpoch);
}
```